src/plugins/template/backend/ops/multiply.cpp

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/reference/multiply.hpp"
+
+#include "evaluate_node.hpp"
+
+template <ov::element::Type_t ET>
+bool evaluate(const std::shared_ptr<ov::op::v1::Multiply>& op, ov::TensorVector& outputs, const ov::TensorVector& inputs) {
+    using T = typename ov::element_type_traits<ET>::value_type;
+    ov::reference::multiply<T>(inputs[0].data<T>(), inputs[1].data<T>(), outputs[0].data<T>(), ov::shape_size(inputs[0].get_shape()));
+    return true;
+}
+
+template <>
+bool evaluate_node<ov::op::v1::Multiply>(std::shared_ptr<ov::Node> node,
```

Comment:
The `v1::Multiply` has `evalute` function and support some types [see](https://github.com/openvinotoolkit/openvino/blob/3238290df5ecb9e0eb0c4254d20bf668c3bc1408/src/core/src/op/multiply.cpp#L62).
The template implementation can just   and this implementation can be just extension of required types not all are required (remove duplicates)

The types such as `i4`, `u4`, `u1` can be removed as reference will not support them correctly.
The `boolean` type can be removed as is not legal value for this operator.

Consider remove types which duplicate list in `ov::op::v1::Multiply::has_evaluate` and which cannot be handled correctly by reference implementation.

Reply:
@praasz 
Thank you very much for your code review! I’ll remove all the duplicates in the next hours.